### PR TITLE
Use Brace Lists to initialize objects.

### PR DESCRIPTION
**Pull request by AtariDreams**
This is for consistency with the rest of the code.

### CODE REVIEW

**Comment by StephanTLavavej** [comment]
Since we're constructing temporaries, I would prefer to see `path{ARGS}`, `pos_type{ARGS}`, and `locale{ARGS}`, instead of just `{ARGS}`. This is a case where I feel that writing a bit more than the compiler demands is justified, because it highlights the construction of the temporary.

In contrast, plain `return { STUFF };` is reasonable when the return type is pair/tuple-like, because the construction of the temporary is not especially interesting or expensive. (For example, range algorithms have return types like this.)

File: stl/inc/strstream [hunk 1]
```diff
@@ -242,7 +242,7 @@ protected:
             || (_Which & ios_base::out && !pptr())) { // N4727 [depr.strstreambuf.virtuals]/14:
                                                       // For a sequence to be positioned, if its next pointer
                                                       // is a null pointer, the positioning operation fails.
-            return pos_type(off_type(-1));
+            return {off_type(-1)};
```
Comment (frederick-vs-ja) [inline stl/inc/strstream:245]: I suspect we can just return `-1LL` (ditto for other `pos_type(off_type(-1))`). IIUC both `pos_type` and `off_type` here are `long long`.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I've pushed a merge with `main` to resolve a simple conflict with a comment change, followed by commits to address code review feedback (as I mentioned, we think that mentioning the types would be a good idea here) and update more occurrences of these types. This significantly expands the size of the diff, but it is still primarily focused on `path`, `locale`, and `pos_type` (plus a couple of things in their immediate vicinity); I tried not to make this PR excessively sprawling, and instead simply tried to complete its improvements for consistency.

To summarize, the changes are:

* Update all occurrences of `path`, `locale`, and `pos_type` to use braces when constructing temporaries (or allocating objects).
* Drop unnecessary parentheses in two successive lines that were constructing `path`s.
* Also update `file_status` and `off_type` to use braces. (The latter is ultimately `long long`, so `off_type(-1)` was a functional-style cast from `int` to `long long`.)
* We were inconsistent about whether to construct `pos_type` (i.e. `fpos`) directly from `-1`, or from an `off_type`. I've chosen to standardize on the full `pos_type{off_type{-1}}` form, as this more closely aligns with what the Standard depicts (and ultimately there is no codegen impact).

Note: I have deliberately not audited the test code. Our test suite is relatively vast and contains lots of old code. We don't spend effort on modernizing its patterns except in unusually important cases (e.g. braces for control flow); also it is sometimes important for test code to use unusual patterns so by default it's better not to mess with them too much. We try to keep the product code more modern and consistent.

**Comment by StephanTLavavej** [comment]
I'm *speculatively* mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for these consistency improvements! :chart_with_upwards_trend: :smile_cat: :tada:

**Pull request merged**